src/main/java/org/sonar/plugins/stash/issue/StashDiffReport.java

```diff
@@ -35,27 +35,40 @@ public void add(StashDiffReport report) {
     diffs.addAll(report.getDiffs());
   }
 
-  private static boolean includeVicinityIssuesForDiff(StashDiff diff, long destination, int range) {
+  private static boolean inVicinityOfChangedDiff(StashDiff diff, long destination, int range) {
     if (range <= 0) {
       return false;
     }
     return Range.closed(diff.getSource() - range, diff.getDestination() + range).contains(destination);
   }
 
+  private static boolean isChangedDiff(StashDiff diff, long destination) {
+    return diff.getDestination() == destination;
+  }
+
   public IssueType getType(String path, long destination, int vicinityRange) {
+    boolean isInContextDiff = false;
     for (StashDiff diff : diffs) {
       if (Objects.equals(diff.getPath(), path)) {
         // Line 0 never belongs to Stash Diff view.
         // It is a global comment with a type set to CONTEXT.
         if (destination == 0) {
           return IssueType.CONTEXT;
-        } else if (destination == diff.getDestination() || includeVicinityIssuesForDiff(diff,
-            destination, vicinityRange)) {
+        } else if (!lineIsChangedDiff(diff)) {
+          // We only care about changed diff
+          continue;
+        } else if (isChangedDiff(diff, destination)) {
           return diff.getType();
+        } else if (inVicinityOfChangedDiff(diff, destination, vicinityRange)) {
+          isInContextDiff = true;

```

Comment:
It should be possible to just `return IssueType.CONTEXT` here and get rid of the state variable and the ternary operator at the end, no?